pipelines/test/tw/gem-check.yaml

```diff
@@ -0,0 +1,19 @@
+name: gem-check
+needs:
+  packages:
+    - gem-check
+
+inputs:
+  packages:
+    description: package(s) containing a ruby gem
+    required: false
+    default: "${{context.name}}"
+  require:
+    description: optional string of exact gem name to require
+    required: false
+    default: ""
```

Comment:
I think the idea that I shared about 2 pipelines fits well here.
We can have one pipeline `gem-check-package` that then just does nothing but checks the well-named package import.

```yaml
  - uses: test/tw/gem-check-package
```

and then one that checks specific requires
```yaml
  - uses: test/tw/ruby-require
    with: |
        name1
        name2
```

mostly interested in thoughts there.

Reply:
Resolving this with an embedded comment.  Actually, I don't believe that we should generally depend on ruby or gem here, as the package itself will sometimes need to depend on a specific version of ruby, etc.  And the test should fail if the alignment is not correct.